docs/docs/architecture/secrets.md

```diff
@@ -35,3 +36,64 @@ The workload owner can derive the workload secret themselves, since it's derived
 If the data owner and the workload owner is the same entity, then they can safely use the workload secrets.
 
 :::
+
+### Secure persistence
+
+Remember that persistent volumes from the cloud provider are untrusted.
+Using the workload secret, applications can set up trusted storage on top of untrusted block devices.
+The following, slightly abbreviated resource outlines how this could be realized:
+
+```yaml
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: volume-tester
+spec:
+  template:
+    spec:
+      containers:
+      - name: main
+        image: my.registry/my-image@sha256:0123...
+        command:
+        - /bin/sh
+        - -ec
+        - | # <-- Custom script that mounts the encrypted disk and then calls the original application.
+          device=/dev/csi0
+          if ! cryptsetup isLuks $device; then
+            cryptsetup luksFormat $device /contrast/secrets/workload-secret-seed
```

Comment:
Could this destroy data if `cryptsetup isLuks` fails for another reason?

Reply:
Yes, it would. Let me add a disclaimer to that.